components/script/dom/response.rs

```diff
@@ -412,6 +413,45 @@ impl Response {
         *self.url.borrow_mut() = Some(final_url);
     }
 
+    pub fn set_url_list(&self, new_url_list: Vec<ServoUrl>) {
+        *self.url_list.borrow_mut() = new_url_list;
+    }
+
+    pub fn set_status(&self, status: Option<StatusCode>) {
+        *self.status.borrow_mut() = status;
+    }
+
+    pub fn set_body(&self, body: Vec<u8>) {
```

Comment:
We shouldn't need these new setter methods anymore.

Reply:
Doesn't that mean the members will be modified directly in different places? I thought this is a more cleaner approach.

Reply:
They're all private members that are modified in the same function. I don't see any benefit to modifying them by calling a setter instead of modifying them directly.

Reply:
In case, we have to do something similar to what we did with `set_type`, these setter is may be a better approach. For eg., say whenever a status is set, the status_message should also be set.

Reply:
If that case comes up, we can consider refactoring at that point.